rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -428,6 +514,130 @@ def spend_asset(
         self.reduce_asset_amount(asset=asset, amount=amount, timestamp=timestamp)
         return None
 
+    def calculate_spend_average_cost_basis(
```

Comment:
This is a clear case of repetition -- will be refactored in the next iteration. 
The reason for this is to be sure that the implementation works and is acceptable.